### DISCUSSION TITLE
Importing 30MB+ JSON records

### DISCUSSION THREAD

**Discussion by marklit** [duckdb/duckdb#8604] [General]
I have a JSONL file that is ~1 GB uncompressed. It has ~1K lines. The dictionaries in each record have around 50-60 keys and most are short strings or integers but there is one field called geom which is a WKB string. These strings are upwards of 30MB each.

What is the most RAM-friendly way to import a JSONL file like this into DuckDB? I'm running v0.8.1-dev327 57912b60a6.

```sql
CREATE OR REPLACE TABLE funding AS
    SELECT *
    FROM read_ndjson_auto('data.jsonl',
                          maximum_object_size=999999999);
```

```
Error: Out of Memory Error: failed to allocate data of size 1.9GB (54.0GB/54.0GB used)
```

**Comment by Alex-Monahan** [comment]
One thing to try would be to specify the schema ahead of time, or maybe create the table with a `limit 100` and use that to determine the schema
